tools/src/main/java/org/apache/kafka/tools/consumer/group/ConsumerGroupCommand.java

```diff
@@ -619,7 +619,11 @@ private Collection<PartitionAssignmentState> describePartitions(
                     consumerIdOpt, hostOpt, clientIdOpt, logEndOffsetOpt, leaderEpoch);
             };
 
-            return offsetsUtils.getLogEndOffsets(topicPartitions).entrySet().stream().map(logEndOffsetResult -> {
+            Set<TopicPartition> nonLeaderTopicPartitions = filterNoneLeaderPartitions(topicPartitions);
+
+            // prepare data for partitions with leaders
+            topicPartitions.removeAll(nonLeaderTopicPartitions);
```

Comment:
we don't need to modify the input `topicPartitions` if `filterNoneLeaderPartitions` returns the partitions having leaders.